File: Complaint Management System Frontend/src/constants/securityPolicy.js
export const LOCKOUT_THRESHOLD = 6;
export const LOCKOUT_DURATION_MINUTES = 30;
export const SESSION_TIMEOUT_MINUTES = 15;
export const SESSION_TIMEOUT_MS = SESSION_TIMEOUT_MINUTES * 60 * 1000;
export const PASSWORD_MIN_LENGTH = 12;

export const PASSWORD_POLICY = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d).{12,}$/;

export const PASSWORD_REQUIREMENTS_MESSAGE =
  'Password must be at least 12 characters and contain uppercase, lowercase, and numeric characters.';

export const PASSWORD_HISTORY_MESSAGE = 'You cannot reuse any of your last 4 passwords.';

export const ACCOUNT_LOCKED_MESSAGE =
  'Your account has been locked due to multiple failed login attempts. Please try again after 30 minutes or contact the system administrator.';

export const PASSWORD_EXPIRED_MESSAGE =
  'Your password has expired. Please change your password to continue.';

export const SESSION_TIMEOUT_MESSAGE =
  'Your session has expired due to inactivity. Please login again.';

export const SESSION_MESSAGE_KEY = 'cms.sessionMessage';

export function passwordExpiryBannerText(daysRemaining) {
  if (daysRemaining <= 1) {
    return 'Your password will expire tomorrow.';
  }
  return `Your password will expire in ${daysRemaining} days.`;
}

File: complaint-management-system/src/main/java/com/dashenbank/cms/security/SecurityPolicy.java
package com.dashenbank.cms.security;

import java.util.regex.Pattern;

public final class SecurityPolicy {

    public static final int LOCKOUT_THRESHOLD = 6;
    public static final int LOCKOUT_DURATION_MINUTES = 30;
    public static final int SESSION_TIMEOUT_MINUTES = 15;
    public static final int PASSWORD_MIN_LENGTH = 12;
    public static final int PASSWORD_HISTORY_COUNT = 4;
    public static final int PASSWORD_EXPIRY_DAYS = 90;
    public static final int PASSWORD_WARNING_DAYS = 10;

    public static final int SESSION_TIMEOUT_MS = SESSION_TIMEOUT_MINUTES * 60 * 1000;

    public static final String PASSWORD_REQUIREMENTS_MESSAGE =
            "Password must be at least 12 characters and contain uppercase, lowercase, and numeric characters.";
    public static final String PASSWORD_HISTORY_MESSAGE =
            "You cannot reuse any of your last 4 passwords.";
    public static final String ACCOUNT_LOCKED_MESSAGE =
            "Your account has been locked due to multiple failed login attempts. "
                    + "Please try again after 30 minutes or contact the system administrator.";
    public static final String PASSWORD_EXPIRED_MESSAGE =
            "Your password has expired. Please change your password to continue.";
    public static final String SESSION_TIMEOUT_MESSAGE =
            "Your session has expired due to inactivity. Please login again.";

    static final Pattern PASSWORD_PATTERN = Pattern.compile(
            "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d).{" + PASSWORD_MIN_LENGTH + ",}$");

    private SecurityPolicy() {
    }

    public static boolean meetsPasswordPolicy(String password) {
        return password != null && PASSWORD_PATTERN.matcher(password).matches();
    }
}
